app/scripts/components/common/mapbox/layers/raster-timeseries.tsx

```diff
@@ -337,92 +337,123 @@ export function MapLayerRasterTimeseries(props: MapLayerRasterTimeseriesProps) {
   );
+
   useEffect(
     () => {
-      let layers: AnyLayer[] = [];
-      let sources: Record<string, AnySourceImpl> = {};
-
-      if (mosaicUrl) {
-        const tileParams = qs.stringify(
-          {
-            assets: 'cog_default',
-            ...sourceParams
-          },
-          // Temporary solution to pass different tile parameters for hls data
-          { arrayFormat: id.toLowerCase().includes('hls') ? 'repeat' : 'comma' }
-        );
-
-        const mosaicSource: RasterSource = {
-          type: 'raster',
-          url: `${mosaicUrl}?${tileParams}`
-        };
-
-        const mosaicLayer: RasterLayer = {
-          id: id,
-          type: 'raster',
-          source: id,
-          layout: {
-            visibility: isHidden ? 'none' : 'visible'
-          },
-          paint: {
-            'raster-opacity': Number(!isHidden),
-            'raster-opacity-transition': {
-              duration: 320
+      const controller = new AbortController();
+
+      async function run() {
+        let layers: AnyLayer[] = [];
+        let sources: Record<string, AnySourceImpl> = {};
+
+        if (mosaicUrl) {
+          const tileParams = qs.stringify(
+            {
+              assets: 'cog_default',
+              ...sourceParams
+            },
+            // Temporary solution to pass different tile parameters for hls data
+            {
+              arrayFormat: id.toLowerCase().includes('hls') ? 'repeat' : 'comma'
             }
-          },
-          minzoom: minZoom,
-          metadata: {
-            layerOrderPosition: 'raster'
+          );
+
+          const tilejsonUrl = `${mosaicUrl}?${tileParams}`;
+
+          let tileServerUrl: string | undefined = undefined;
+          try {
+            const tilejsonData = await requestQuickCache({
+              url: tilejsonUrl,
+              method: 'GET',
+              payload: null,
+              controller
+            });
+            tileServerUrl = tilejsonData.tiles[0];
+          } catch (error) {
+            // Ignore errors.
           }
-        };
 
-        sources = {
-          ...sources,
-          [id]: mosaicSource
-        };
-        layers = [...layers, mosaicLayer];
-      }
+          const mosaicSource: RasterSource = {
+            type: 'raster',
+            url: tilejsonUrl
+          };
+
+          const mosaicLayer: RasterLayer = {
+            id: id,
+            type: 'raster',
+            source: id,
+            layout: {
+              visibility: isHidden ? 'none' : 'visible'
+            },
+            paint: {
+              'raster-opacity': Number(!isHidden),
+              'raster-opacity-transition': {
+                duration: 320
+              }
+            },
+            minzoom: minZoom,
+            metadata: {
+              id,
+              layerOrderPosition: 'raster',
+              xyzTileUrl: tileServerUrl,
+              wmtsTileUrl: `${mosaicUrl}/WMTSCapabilities.xml?${tileParams}`
```

Comment:
@anayeaye can you confirm this is the right format for wmts tile URL ?

Reply:
I'm not sure: if `mosaicUrl` = `/mosaic/{searchid}/{TileMatrixSetId}` or `/mosaic/{searchid}` this works. 

If mosaicUrl includes `/tiles/` like `/mosaic/tiles/{searchid}/...`, it won't work (which is generally how the XYZ urls are formatted).

Reply:
Just looked at the preview `tiles.json` needs to be stripped from `/mosaic/cfa5568f19003ffe39f4f05c9a38b481/tilejson.json/WMTSCapabilities.xml?`